packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -0,0 +1,244 @@
+
+    const focussedCell = this.grid.rows[this.focussedRow].cells[this.focussedCol];
+    focussedCell.setAttribute('tabIndex', '0');
+
+    // add event listeners to handle user key events in the grid.
+    this.grid.addEventListener('keydown', this.handleKeyDown);
+    this.grid.addEventListener('click', this.handleClick);
+  }
+
+  handleKeyDown(event) {
+    if (event.target?.parentNode?.parentNode?.parentNode?.id !== this.props.id) {
+      return;
+    }
+
+    const currentFocussedRow = event.target.parentElement.rowIndex;
+    const currentFocussedCol = event.target.cellIndex;
+
+    let nextRow = currentFocussedRow;
+    let nextCol = currentFocussedCol;
+
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_UP:
+        nextRow -= 1;
+        break;
+      case KeyCode.KEY_DOWN:
+        nextRow += 1;
+        break;
+      case KeyCode.KEY_LEFT:
+        nextCol -= 1;
+        break;
+      case KeyCode.KEY_RIGHT:
+        nextCol += 1;
+        break;
+      case KeyCode.KEY_HOME:
+        nextCol = 0;
+        if (event.ctrlKey) {
+          nextRow = this.getNumberOfHeaderRows();
+        }
+        break;
+      case KeyCode.KEY_END:
+        nextCol = this.props.rows[0].cells.length - 1;
+        if (event.ctrlKey) {
+          nextRow = (this.props.rows.length - 1) + this.getNumberOfHeaderRows();
+        }
+        break;
+      default:
+        return;
+    }
+    if (nextRow >= this.grid.rows.length || nextCol >= this.grid.rows[0].cells.length) {
+      event.preventDefault();
+      return;
+    }
+    if (nextCol < 0 || nextRow < 0) {
+      event.preventDefault();
+      return;
+    }
+
+    // Remove the ability to receive visible focus as only one cell should be able to receive tab focus.
+    // This is also how tabbing back to the grid ensures the grid last cell with the focus will be re-entered.
+    this.grid.rows[currentFocussedRow].cells[currentFocussedCol].setAttribute('tabIndex', '-1');
+
+    this.focussedRow = nextRow;
+    this.focussedCol = nextCol;
+
+    const elementWithTabStop = this.grid.rows[nextRow].cells[nextCol];
+    elementWithTabStop.setAttribute('tabIndex', '0');
+    elementWithTabStop.focus();
+    event.preventDefault();
+  }
+
+  handleClick(event) {
+    // Is this event for something that is on the grid
+    if (event.target?.parentNode?.parentNode?.parentNode?.id !== this.props.id) {
+      return;
+    }
+    // Remove Tab stop from previous cell in table that has focus and set it to the cell that was clicked.
+    this.grid.rows[this.focussedRow].cells[this.focussedCol].setAttribute('tabIndex', '-1');
+    this.focussedRow = event.target.parentElement.rowIndex;
+    this.focussedCol = event.target.cellIndex;
+    const clickedCell = event.target;
+    clickedCell.setAttribute('tabIndex', '0');
+    clickedCell.focus();
+    event.preventDefault();
+  }
+
+  getNumberOfHeaderRows() {
+    return this.props.columnHeaders?.pinnedColumns?.length > 0 ? 1 : 0;
```

Comment:
Why would pinned columns be a property of the column headers?  Whether or not a column is pinned should be part of the columns themselves.

Reply:
Pinned/overflow columns are not yet implemented. We have restructured the data to only consider columns at this time.